api/app/src/main/kotlin/packit/model/User.kt

```diff
@@ -17,20 +18,42 @@ class User(
         inverseJoinColumns = [JoinColumn(name = "role_id")]
     )
     var roles: MutableList<Role> = mutableListOf(),
-    var disabled: Boolean,
+    var disabled: Boolean = false,
     val userSource: String,
-    val displayName: String?,
+    val displayName: String? = null,
     val email: String? = null,
     var password: String? = null,
     var lastLoggedIn: Instant? = null,
     @Id
     @GeneratedValue(strategy = GenerationType.UUID)
     val id: UUID? = null,
     @OneToMany(mappedBy = "user")
-    var runInfos: MutableList<RunInfo> = mutableListOf()
+    var runInfos: MutableList<RunInfo> = mutableListOf(),
+
+    @OneToOne()
+    @JoinColumn(
+        name = "username",
+        referencedColumnName = "name",
+        insertable = false,
+        updatable = false
+    )
+    @Filter(name = "isUsername", condition = "isUsername = TRUE")
+    var userRole: Role? = null,
```

Comment:
I'm honestly not all that sure of what I'm doing here, but it seems to be working. I wanted the `userRole` to be available from within the `toDto` methods and a join seemed like the cleanest way.